### PR TITLE
OAK-10914 : use negative lookahead instead of negation for main fullgc query

### PR DESCRIPTION
**Pull request by stefan-egli**

### CODE REVIEW

**Comment by stefan-egli** [comment]
note that the `hint` was still required to force it to use the `modified-id` index in the first place

**Review by rishabhdaim — APPROVED** [review]

**Comment by stefan-egli** [comment]
I'll let it do the tests before merging (they will likely fail - but shouldn't have any regressions at least)

**Review by Joscorbe — APPROVED** [review]

**Comment by stefan-egli** [comment]
the `branch` jenkins one had flaky errors it seems - should not block this PR though
```[ERROR] Errors: 

[ERROR]   CacheWarmingTest.prefetch:139->newMongoDocumentStore:109 » MongoTimeout Timed ...

[ERROR]   CacheWarmingTest.simple_cleanCaches_noprefetch:124->doSimple:175->newMongoDocumentStore:109 » MongoTimeout

[ERROR]   CacheWarmingTest.simple_cleanCaches_withprefetch:129->doSimple:175->newMongoDocumentStore:109 » MongoTimeout

[ERROR]   CacheWarmingTest.simple_nocleanCaches_noprefetch:119->doSimple:175->newMongoDocumentStore:109 » MongoTimeout

[ERROR]   CacheWarmingTest.simple_nocleanCaches_withprefetch:134->doSimple:175->newMongoDocumentStore:109 » MongoTimeout

[INFO] 
```

**Pull request merged**
